packages/worker-shared/src/sentry.ts

```diff
@@ -0,0 +1,34 @@
+import { WorkerVersionMetadata } from '@cloudflare/workers-types'
+import { Toucan } from 'toucan-js'
+import { requiredEnv } from './env'
+
+interface Context {
+	waitUntil: ExecutionContext['waitUntil']
+	request?: Request
+}
+
+interface SentryEnvironment {
+	readonly SENTRY_DSN: string | undefined
+	readonly WORKER_NAME: string | undefined
+	readonly CF_VERSION_METADATA: WorkerVersionMetadata
+}
+
+export function createSentry(ctx: Context, env: SentryEnvironment, request?: Request) {
+	const { SENTRY_DSN, WORKER_NAME, CF_VERSION_METADATA } = requiredEnv(env, {
+		SENTRY_DSN: true,
+		WORKER_NAME: true,
+		CF_VERSION_METADATA: true,
+	})
+
+	return new Toucan({
+		dsn: SENTRY_DSN,
+		release: `${WORKER_NAME}.${CF_VERSION_METADATA.id}`,
```

Comment:
i think this release name doesn't match with the one used when uploading the source maps now that it includes the full domain. I think we can set up the WORKER_ID in the deploy scripts and pass that through to `wranglerDeploy` to use when it creates the sentry release and uploads the source maps, so that it doesn't need to extract it from either the wrangler output or the wrangler.toml

Reply:
The release name that we use when uploading source maps is `${workerNameMatch[1]}.${versionMatch[1]}`, not the full domain i think?

Reply:
ah sorry yeah you're right, i misread the code